pkg/gitnessclient/gitness_client.go

```diff
@@ -0,0 +1,416 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitnessclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+const personalNamespaceId = "<PERSONAL>"
+
+type GitnessClient struct {
+	Token   string
+	BaseURL *url.URL
+}
+
+func NewGitnessClient(token string, baseUrl *url.URL) *GitnessClient {
+	return &GitnessClient{
+		Token: token,
+		BaseURL: &url.URL{
+			Scheme: "http",
+			Host:   baseUrl.Host,
+			Path:   baseUrl.Path,
+		},
+	}
+}
+
+func (g *GitnessClient) GetSpaceAdmin(spaceName string) (*apiSpaceMemberResponse, error) {
+
```

Comment:
Please remove newlines from the begging of functions. I can see they were left at multiple places. This is to ensure consistency in Daytona's codebase.